### PR TITLE
add pagination for queries using havings

### PR DESCRIPTION
**Pull request by mjauvin**
Apply Laravel 7.x PR to fix this:

ref. https://github.com/laravel/framework/pull/32688/

Thanks to @alvaro-canepa and @acasar for this!

### CODE REVIEW

File: src/Database/QueryBuilder.php [hunk 1]
```diff
@@ -433,4 +434,40 @@ public function selectConcat(array $parts, string $as)
...
+            // We don't need simple columns, only specials like subselects which is why
+            // we're using havings after all.
+            foreach ($query->columns as $key => $value) {
+                if (is_string($value)) {
+                    unset($query->columns[$key]);
+                }
+            }
+
+            $countQuery = Db::connection($this->getConnection()->getName())->table(
+                Db::raw('('.$query->toSql().') as x')
```
Comment (LukeTowers) [inline src/Database/QueryBuilder.php]: Any particular reason for doing `as x`? I feel like that is highly likely to conflict with developer's own custom select statements.

Comment (mjauvin) [reply src/Database/QueryBuilder.php]: No idea @LukeTowers , I just used the code @alvaro-canepa took from the other laravel package implementation.

**Comment by alvaro-canepa** [comment]
https://github.com/justbetter/laravel-pagination-with-havings
This is the original repo.

**Comment by alvaro-canepa** [comment]
I tested with `withCount` + `having` + `pagination` and works fine!
```php
    public function scopeClosed($obQuery)
    {
        return $obQuery->withCount(['orders', 'items'])
            ->having('orders_count', '=', \DB::raw('items_count'));
    }
```

**Comment by LukeTowers** [comment]
Looks like it was added directly to Laravel in 7.x. Would be better to use that version as a base instead: https://github.com/laravel/framework/pull/32624.

Also, looks like @acasar had the solution all along, great job man! :)

**Comment by mjauvin** [comment]
This version was merged, the other one had errors:
https://github.com/laravel/framework/pull/32688

**Comment by mjauvin** [comment]
Now I need to add the change to `$this->groups`

**Comment by mjauvin** [comment]
@alvaro-canepa can you test this new implementation to make sure it still work as expected?

**Comment by mjauvin** [comment]
@alvaro-canepa did you get a chance to test this?

**Comment by mjauvin** [comment]
please keep active.

**Comment by alvaro-canepa** [comment]
> @alvaro-canepa can you test this new implementation to make sure it still work as expected?

@mjauvin  Sorry, I haven't had a chance to try it yet!

**Pull request closed without merging**